### PR TITLE
Perform graceful stops on PHP-FPM

### PR DESCRIPTION
**Pull request by lcobucci**
The default signal used by docker to stop a container is `SIGTERM` and that signal is understood by PHP-FPM as immediate termination.

This means that `php-fpm` won't wait for in-flight requests to finish before of stopping the workers and main process. This behaviour is quite undesired for production environments and should be avoided as much as possible.

This implementation is meant to be extremely simple and only address the stop signal used when running `docker stop`. `php-fpm` signal choices are somewhat peculiar (`SIGUSR2` for example) and if we want to "fix" them we would have to use a tool (e.g.: `dumb-init`).

More info:

 - https://linux.die.net/man/8/php-fpm
 - https://docs.docker.com/engine/reference/builder/#stopsignal
 - https://github.com/Yelp/dumb-init
 - https://github.com/usabilla/php-docker-template/pull/102

### CODE REVIEW

**Comment by lcobucci** [comment]
@renatomefi yeap, forgot that :sweat_smile: thanks

**Comment by lcobucci** [comment]
@yosifkit the build failure looks completely unrelated to the changes but please let me know if I have to do something :+1:

File: fpm-Dockerfile-block-2 [hunk 1]
```diff
@@ -39,4 +39,5 @@ RUN set -ex \
 	} | tee php-fpm.d/zz-docker.conf
 
 EXPOSE 9000
+STOPSIGNAL SIGQUIT
```
Comment (tianon) [inline fpm-Dockerfile-block-2]: I have searched and cannot find this officially documented anywhere on https://php.net -- it appears the man page really is the only place it's officially noted. 😓

Can we update this to link to https://github.com/php/php-src/blob/17baa87faddc2550def3ae7314236826bc1b1398/sapi/fpm/php-fpm.8.in#L163 in a code comment for future reference?

ala:

```dockerfile
  ... tee ...conf

# https://github.com/php/php-src/blob/17baa87faddc2550def3ae7314236826bc1b1398/sapi/fpm/php-fpm.8.in#L163
STOPSIGNAL SIGQUIT

EXPOSE 9000
CMD ...
```

Comment (renatomefi) [reply fpm-Dockerfile-block-2]: Great digging tho! The man page seems to be accurate at least!

Comment (lcobucci) [reply fpm-Dockerfile-block-2]: @tianon great digging indeed! I'll add it :+1:

**Comment by adri** [comment]
Great timing, thanks for this :)

**Comment by lcobucci** [comment]
@yosifkit @tianon do we need something else here?

**Comment by tianon** [comment]
Sorry for the delay! :+1:

**Comment by lcobucci** [comment]
@tianon no problem! Thanks 👍

**Pull request merged**